server/src/main/java/org/elasticsearch/index/query/GeoShapeQueryBuilder.java

```diff
@@ -815,10 +815,10 @@ protected QueryBuilder doRewrite(QueryRewriteContext queryRewriteContext) throws
                     getRequest = new GetRequest(indexedShapeIndex, indexedShapeType, indexedShapeId);
                 }
                 getRequest.routing(indexedShapeRouting);
-                fetch(client, getRequest, indexedShapePath, ActionListener.wrap(builder-> {
+                fetch(client, getRequest, indexedShapePath, ActionListener.map(listener, builder -> {
                     supplier.set(builder);
-                    listener.onResponse(null);
-                }, listener::onFailure));
+                    return null;
```

Comment:
With side-effect, I think ActionListener.wrap is more suitable than the new method.